internal/dag/policy.go

```diff
@@ -188,6 +190,54 @@ func headersPolicyRoute(policy *contour_api_v1.HeadersPolicy, allowHostRewrite b
 	}, nil
 }
 
+// headersPolicyGatewayAPI builds a *HeaderPolicy for the supplied HTTPRequestHeaderFilter.
+// TODO: Take care about the order of operators once https://github.com/kubernetes-sigs/gateway-api/issues/480 was solved.
+func headersPolicyGatewayAPI(hf *gatewayapi_v1alpha1.HTTPRequestHeaderFilter) (*HeadersPolicy, error) {
```

Comment:
I think we should refactor the headersPolicy bits that exist today into something more generic such that we don't duplicate this logic. We could do that work on a separate PR, or refactor after this one.

Reply:
I think refactor after this PR by separate PR is better. I will open an issue ticket for it.

Reply:
Opened https://github.com/projectcontour/contour/issues/3562

Reply:
👍